cartography/models/azure/tags/storage_tag.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+
+from cartography.models.azure.tags.tag import AzureTagSchema
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.relationships import CartographyRelProperties
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class StorageAccountToTagRelProperties(CartographyRelProperties):
+    lastupdated: PropertyRef = PropertyRef("lastupdated", set_in_kwargs=True)
+
+
+@dataclass(frozen=True)
+class StorageAccountToTagRel(CartographyRelSchema):
+    target_node_label: str = "AzureStorageAccount"
+    target_node_matcher: TargetNodeMatcher = make_target_node_matcher(
+        {"id": PropertyRef("resource_id")},
+    )
+    direction: LinkDirection = LinkDirection.INWARD
+    rel_label: str = "TAGGED"
+    properties: StorageAccountToTagRelProperties = StorageAccountToTagRelProperties()
+
+
+@dataclass(frozen=True)
+class AzureStorageTagsSchema(AzureTagSchema):
+    """
+    Inherits standard tag properties from AzureTagSchema and adds the
+    specific relationship to AzureStorageAccount.
+    """
+
+    other_relationships: OtherRelationships = OtherRelationships(
```

Comment:
Let's reverse this. Instead of defining the relationship on the tag to the storage account, let's define the relationship on the storage account to the tag. This way, as the code grows, the tag won't need to have multiple relationships defined on it, making the ingestion query inefficient